heroku_connect/models.py

```diff
@@ -0,0 +1,342 @@
+    IGNORED = 'IGNORED'
+    """No sync attempted. (Usually because it's not needed.)"""
+    FAILED = 'FAILED'
+    """Sync attempt failed. Check `sf_message`."""
+    READONLY = 'READONLY'
+    """Captured update on read-only table."""
+    NEW = 'NEW'
+    """Newly captured change reedy for processing."""
+    IGNORE = 'IGNORE'
+    """Newly captured change that needs no syncing."""
+    PENDING = 'PENDING'
+    """Currently being processed."""
+
+    REQUEUE = 'REQUEUE'
+    """Marks a archived entry to be copied back as NEW into the current log."""
+    REQUEUED = 'REQUEUED'
+    """An archived entry that was copied back into as NEW into the current log."""
+
+    @classmethod
+    def choices(cls):
+        return tuple((getattr(cls, name), name) for name in dir(cls) if name.isupper())
+
+
+class TriggerLogAbstract(models.Model):
+    """Support for accessing the Heroku Connect Trigger Log data and related actions.
+
+    Heroku Connect uses a Trigger Log table to track local changes to connected models (that is,
+    in the Heroku database. Such changes are recorded as rows in the trigger log and, for
+    read-write mappings, eventually written back to Salesforce.
+
+    Old logs are moved to an archive table (usually after 24 hours), from where they are purged
+    eventually (currently 30 days for paid plans, 7 days for demo). Recent logs are modeled by
+    :class:`TriggerLog`; archived logs by :class:`TriggerLogArchive`.
+
+    The data represented by these models is maintained entirely by Heroku Connect, and is
+    instrumental to its operations; it should therefore not be modified. A possible exception is
+    the ``state`` field, which may be changed as detailed in the `error handling`_ section in the
+    Heroku Connect documentation.
+
+    .. seealso::
+        - :class:`.TriggerLogQuerySet`
+        - `Trigger Log in Heroku Connect docs`_
+
+    .. _Trigger Log in Heroku Connect docs:
+        https://devcenter.heroku.com/articles/writing-data-to-salesforce-with-heroku-connect#understanding-the-trigger-log
+    .. _error handling:
+        https://devcenter.heroku.com/articles/writing-data-to-salesforce-with-heroku-connect#write-errors
+
+    """
+
+    # read-only fields
+    id = models.BigIntegerField(primary_key=True, editable=False)
+    created_at = models.DateTimeField(editable=False, null=True)
+    updated_at = models.DateTimeField(editable=False, null=True)
+    processed_at = models.DateTimeField(editable=False, null=True)
+    table_name = models.CharField(max_length=128, editable=False)
+    record_id = models.BigIntegerField(editable=False)
+    sf_id = models.CharField(max_length=18, editable=False, null=True, db_column='sfid')
+    action = models.CharField(max_length=7, editable=False, choices=TriggerLogAction.choices())
+    sf_message = models.TextField(editable=False, null=True, blank=True)
+
+    # TODO: these are more useful as HStoreFields (if 'django.contrib.postgres' in INSTALLED_APPS)
```

Comment:
Just document it as a dependency. It's not like you can you any other database with Heroku Connect ;)
Please make sure to add extension to the initial migration:
https://docs.djangoproject.com/en/2.0/ref/contrib/postgres/operations/#hstoreextension
It isn't added automatically :/

Reply:
@codingjoe Almost done addressing the feedback bit by bit, just getting around to this...

> Please make sure to add [hstore] extension to the initial migration:

This will make the migration fail if the database user lacks the appropriate privileges. Maybe it's better to add a check which looks at [`pg_available_extensions`](https://www.postgresql.org/docs/10/static/view-pg-available-extensions.html) and prints instructions to add the extension?